src/emoji/__tests__/data-test.js

```diff
@@ -67,6 +67,28 @@ describe('getFilteredEmojis', () => {
     ]);
   });
 
+  test('returns literal emoji', () => {
+    const list = getFilteredEmojis('🖤', {});
+    expect(list).toEqual([
+      { emoji_type: 'unicode', code: '1f5a4',  name: 'black_heart' },
+    ]);
+  });
+
+  // This is sort of odd behaviour in some sense: the UI is showing the user
+  // multiple emojis, when they can only react with one of them (if they react
+  // twice with :+1: and :thumbs_up:, it'll only be stored as a single
+  // reaction). We could canonicalize the emoji name in cases where there are
+  // multiple aliases, but it seems somewhat useful to allow the user to
+  // discover that the aliases exist, and unlikely to confuse in practice
+  // (since people will likely just pick an arbitrary one and be done with it)
+  test('returns multiple literal emoji', () => {
```

Comment:
Yeah. There's another reason this is the right behavior, too -- it lets the user pick which name they want to use. For example if you enter 🗽 aka U+1F5FD STATUE OF LIBERTY, on a version that has the changes in this PR, you get this:
![image](https://user-images.githubusercontent.com/28173/109371862-9e697300-785b-11eb-9cee-c7ccb266651f.png)
with `:new_york:`, `:statue:`, and `:statue_of_liberty:` as your choices. They all result in reactions with the same `emoji_code` and hence the same glyph 🗽, but the `emoji_name` will be different, and the latter controls what other users will see as the name when (on web/desktop) they hover over the reaction.

(In this particular example, probably the actual right answer is that we should not have added `:statue:` as an alias for STATUE OF LIBERTY, and maybe not `:new_york:` either. But as long as we have aliases where the names have any difference in meaning, we'll want users to be able to choose.)

This is unfortunately not really documented in our emoji subsystem doc:
https://zulip.readthedocs.io/en/latest/subsystems/emoji.html
but you can get some of it from the Reaction class (mostly its AbstractReaction base class) and the comments on it, in the server's models file:
https://github.com/zulip/zulip/blob/15c4a72ac5297b1213ba93e986147c4c27a75af2/zerver/models.py#L2179